File: test_secure.py
from flask import Flask
from flask_fastspring import FastSpring

key = b'0123456789ABCDEF'
payload = b'Hello, world!'

app = Flask(__name__)
app.config['FASTSPRING_STOREFRONT'] = ''
app.config['FASTSPRING_USERNAME'] = ''
app.config['FASTSPRING_PASSWORD'] = ''
app.config['FASTSPRING_PRIVATE_KEY'] = 'test_secure.key'

fastspring = FastSpring(app)
print(fastspring.secure_payload(key, payload))
print(fastspring.secure_key(key))

File: test_secure.php
<?php

$key = '0123456789ABCDEF';
$payload = 'Hello, world!';

$cipher_text = openssl_encrypt($payload, 'AES-128-ECB', $key, OPENSSL_RAW_DATA);
$secure_payload = base64_encode($cipher_text);

$private_key = openssl_pkey_get_private(file_get_contents('test_secure.key'));
openssl_private_encrypt($key, $aes_key_encrypted, $private_key);
$secure_key = base64_encode($aes_key_encrypted);

echo $secure_payload, "\n";
echo $secure_key, "\n";

?>
